ext/java/org/msgpack/jruby/Packer.java

```diff
@@ -30,20 +35,139 @@ public IRubyObject allocate(Ruby runtime, RubyClass type) {
     }
   }
 
+  static class ExtRegistry {
+    private Ruby runtime;
+    public RubyHash hash;
+    public RubyHash cache;
```

Comment:
I think the code below gets really complicated because it uses `RubyHash` where a Java `Map` would work better. The only reason to use `RubyHash` is when you need to return it to Ruby code, but it doesn't look like that happens here.